client/state/data-layer/wpcom-http/index.js

```diff
@@ -26,6 +27,7 @@ const fetcherMap = method => get( {
 export const successMeta = data => ( { meta: { dataLayer: { data } } } );
 export const failureMeta = error => ( { meta: { dataLayer: { error } } } );
 export const progressMeta = ( { size: total, loaded } ) => ( { meta: { dataLayer: { progress: { total, loaded } } } } );
+export const requestKey = ( { path, query } ) => JSON.stringify( { type: WPCOM_HTTP_REQUEST, path, query } );
```

Comment:
w.r.t the solution: since this doesn't need to be generic for _any_ object I think we can do something simpler:

```jsx
const requestKey = action => `{
    type: ${ WPCOM_HTTP_REQUEST },
    path: ${ action.path },
    query: ${ action.query }
}`;
```